controllers/admin_main.php

```diff
@@ -27,12 +39,714 @@ public function index()
+            'name',
+            'id'
+        );
+
+        // Get open invoices
+        $invoices = $this->Form->collapseObjectArray(
+            $this->Invoices->getAll($client->id),
+            'id_value',
+            'id'
+        );
+
+        // Get service statuses
+        $status = $this->Services->getStatusTypes();
+        unset($status['in_review']);
+
+        // Set vars
+        $this->set('domain', $domain);
+        $this->set(
+            'form',
+            $this->partial(
+                'admin_main_add_configuration',
+                compact(
+                    'client',
+                    'action',
+                    'domain',
+                    'package',
+                    'years',
+                    'modules',
+                    'invoices',
+                    'status',
+                    'vars'
+                )
+            )
+        );
+    }
+
+    /**
+     * Confirmation step
+     */
+    private function renderConfirmationStep($client)
+    {
+        // Set action
+        $action = 'register';
+        if (isset($this->post['transfer']) && $this->post['transfer'] == '1') {
+            $action = 'transfer';
+        }
+
+        // Redirect to lookup if no post data has been passed
+        if (empty($this->post) || is_null($action)) {
+            $this->redirect($this->base_uri . 'plugin/domains/admin_main/add/' . $client->id . '/');
+        }
+
+        // Redirect to lookup if no valid domain has been provided
+        $validator = new Server();
+        if (!$validator->isDomain($this->post['domain'] ?? '')) {
+            $this->flashMessage('error', Language::_('AdminMain.!error.unsupported_domain', true), null, false);
+            $this->redirect($this->base_uri . 'plugin/domains/admin_main/add/' . $client->id . '/lookup/');
+        }
+
+        // Set checkboxes
+        $checkboxes = ['use_module', 'notify_order'];
+        foreach ($checkboxes as $checkbox) {
+            if (!isset($this->post[$checkbox])) {
+                $this->post[$checkbox] = 'false';
+            }
+        }
+
+        // Get TLD from domain
+        $domain = $this->post['domain'];
+        $tld = strstr($this->post['domain'] ?? '', '.');
+
+        // Check if a package exists for the given module
+        $package = $this->DomainsTlds->getTldPackageByModuleId(
+            $tld,
+            $this->post['module'],
+            Configure::get('Blesta.company_id')
+        );
+
+        // If a package exists, but it's not active, make it restricted
+        if (isset($package->status) && $package->status == 'inactive') {
+            $this->Packages->edit($package->id, ['status' => 'restricted']);
+        }
+
+        // If a package doesn't exist for this TLD and the provided module, clone the default one
+        if (empty($package)) {
+            $this->DomainsTlds->duplicate($tld, $this->post['module']);
```

Comment:
Still getting an error if it's a registrar that doesn't have an existing package for the TLD.

Reply:
I was not able to reproduce the issue, but I made some changes that may help. If you are still getting the errors after this update, let me know the steps you are following to reproduce it and the modules involved in your testing.

Reply:
So I 
1. Create a new tld like .biz using the generic domains module
2. Go to a client and add a domain
3. Select a .biz domain
4. Select enom as the registrar
5. Continue, and get the error "Undefined property: stdClass::$type"